public/js/reducers/sources.js

```diff
@@ -205,16 +205,12 @@ function getNewSelectedSource(state, id) : ?Source {
 // (right now) to type those wrapped functions.
 type OuterState = { sources: Record<SourcesState> };
 
-function getSource(state: OuterState, id: string) {
-  return state.sources.sources.get(id);
+function getSourceByURL(sources: any, url: string) {
+  return sources.find(source => source.get("url") == url);
 }
 
-function getSourceByURL(state: OuterState, url: string) {
-  return state.sources.sources.find(source => source.get("url") == url);
-}
-
-function getSourceById(state: OuterState, id: string) {
-  return state.sources.sources.find(source => source.get("id") == id);
+function getSourceById(sources: any, id: string) {
```

Comment:
Yep, I agree.
I didn't find a similar function to `debuggerAgent.setBreakpointsActive()` for Firefox. So the idea is to add/remove breakpoints in the Firefox client. As some functions related to sources were used in `client/firefox/commands.js`, the idea was to pass only the sources to those functions rather than the entire state.